Dockerfile

```diff
@@ -11,8 +15,13 @@ RUN conda install -c conda-forge google-cloud-sdk && \
         git gcc rsync sudo patch openssh-server \
         pciutils nano fuse socat netcat-openbsd curl rsync vim tini && \
     rm -rf /var/lib/apt/lists/* && \
-    # Install kubectl
-    curl -LO "https://dl.k8s.io/release/v1.31.6/bin/linux/amd64/kubectl" && \
+    # Install kubectl based on architecture
+    ARCH=$(uname -m) && \
+    if [ "$ARCH" = "x86_64" ]; then \
+        curl -LO "https://dl.k8s.io/release/v1.31.6/bin/linux/amd64/kubectl"; \
```

Comment:
I'm wondering if this can be simplified as `curl -LO "https://dl.k8s.io/release/v1.31.6/bin/linux/${TARGETARCH}/kubectl";`

Reply:
Good point! Simplified the impl and make it more general